R/luz.R

```diff
@@ -0,0 +1,64 @@
+#' @rdname vetiver_create_description
+#' @export
+vetiver_create_description.luz_module_fitted <- function(model) {
+    glue("A luz module")
+}
+
+#' @rdname vetiver_create_meta
+#' @export
+vetiver_create_meta.luz_module_fitted <- function(model, metadata) {
+    pkgs <- c("luz", model$model$required_pkgs)
+    vetiver_meta(metadata, required_pkgs = pkgs)
+}
+
+#' @rdname vetiver_create_ptype
+#' @export
+vetiver_ptype.luz_module_fitted <- function(model, ...) {
+    rlang::check_dots_used()
+    dots <- list(...)
+    check_ptype_data(dots)
+
+    data <- dots$prototype_data
+    data <- tensors_to_array(data)
+
+    # luz doesn't require named inputs. we use `input` if the user didn't provide any.
+    if (!is.list(data)) {
+        data <- tibble::tibble(input = data)
+    }
+
+    ptype <- vctrs::vec_ptype(data)
+    tibble::as_tibble(ptype)
+}
+
+#' @rdname vetiver_create_description
+#' @export
+vetiver_prepare_model.luz_module_fitted <- function(model) {
+    bundle::bundle(model)
+}
+
+#' @rdname handler_startup
+#' @export
+handler_startup.luz_module_fitted <- function(vetiver_model) {
+    attach_pkgs(vetiver_model$metadata$required_pkgs)
+}
+
+#' @rdname handler_startup
+#' @export
+handler_predict.luz_module_fitted <- function(vetiver_model, ...) {
+    force(vetiver_model)
+    function(req) {
+        new_data <- vetiver_type_convert(req$body, vetiver_model$ptype)
+        preds <- asNamespace("vetiver")$tensors_to_array(predict(vetiver_model$model, new_data))
```

Comment:
vetiver should already be loaded where the API is running. Can you tell me more about  accessing the namespace like this?

Reply:
For some reason during initial tests that function couldn't be found, but I probably made typo or something and then thought it was because this was an internal function. Fixed!